Core/include/Acts/Geometry/LayerBlueprintNode.hpp

```diff
@@ -0,0 +1,126 @@
+
+  /// This function participates in the geometry construction.
+  /// It will:
+  /// -# Analyze the surfaces provided and produce a wrapping volume
+  /// -# Register the surfaces with the volume
+  /// -# Return the volume
+  /// @note At least one surfaces needs to be registered via
+  ///       @ref Acts::LayerBlueprintNode::setSurfaces before
+  ///       geometry construction.
+  Volume& build(const BlueprintOptions& options, const GeometryContext& gctx,
+                const Logger& logger = Acts::getDummyLogger()) override;
+
+  /// Register a set of surfaces with the layer node.
+  /// @param surfaces The surfaces to register
+  /// @return Reference to this node for chaining
+  LayerBlueprintNode& setSurfaces(
+      std::vector<std::shared_ptr<Surface>> surfaces);
+
+  /// Access the registered surfaces.
+  /// @return The registered surfaces
+  const std::vector<std::shared_ptr<Surface>>& surfaces() const;
+
+  /// Set the transformation of the layer node.
+  /// This can be used to specifically orient the resulting layer volume.
+  /// @param transform The transformation to set
+  /// @return Reference to this node for chaining
+  LayerBlueprintNode& setTransform(const Transform3& transform);
+
+  /// Access the transformation of the layer node.
+  /// @return The transformation
+  const Transform3& transform() const;
+
+  /// Set the envelope of the layer node. This configures the amount of space to
+  /// add around the contained surfaces.
+  /// @param envelope The envelope to set
+  /// @return Reference to this node for chaining
+  LayerBlueprintNode& setEnvelope(const ExtentEnvelope& envelope);
+
+  /// Access the envelope of the layer node.
+  /// @return The envelope
+  const ExtentEnvelope& envelope() const;
+
+  /// Set the layer type of the layer node.
+  /// @param layerType The layer type to set
+  /// @return Reference to this node for chaining
+  LayerBlueprintNode& setLayerType(LayerType layerType);
+
+  /// Access the layer type of the layer node.
+  /// @return The layer type
+  const LayerType& layerType() const;
+
+ private:
+  /// @copydoc Acts::BlueprintNode::addToGraphviz
+  void addToGraphviz(std::ostream& os) const override;
+
+  /// Helper method that performs the volume creation from the configured
+  /// surfaces. It converts from an @p extent object to an instance of @ref
+  /// Acts::VolumeBounds.
+  /// @param extent The extent to use for the volume creation
+  /// @param logger The logger to use
+  void buildVolume(const Extent& extent, const Logger& logger);
+
+  std::string m_name;
+  std::vector<std::shared_ptr<Surface>> m_surfaces{};
+  Transform3 m_transform = Transform3::Identity();
+  ExtentEnvelope m_envelope = ExtentEnvelope::Zero();
+  LayerType m_layerType = LayerType::Cylinder;
+};
+
+/// Output operator for the layer type enum.
+/// @param os The output stream
+/// @param type The layer type
+std::ostream& operator<<(std::ostream& os, LayerBlueprintNode::LayerType type);
```

Comment:
global scope because the implementation is large?

Reply:
I don't think I can define a hidden friend for an enum. I wasn't sure a hidden friend in `LayerBlueprintNode` would do the trick, but it seems to compile 🤷‍♀️